src/lib.rs

```diff
@@ -14,39 +24,184 @@ fn process_instruction(
         accounts.len(),
         instruction_data
     ));
-    Ok(())
+
+    let account_info_iter = &mut accounts.iter();
+    let program_token_info = next_account_info(account_info_iter)?;
+    let (program_token_address, program_token_bump_seed) =
+        Pubkey::find_program_address(&[br"program-token"], program_id);
+
+    if program_token_address != *program_token_info.key {
+        info!("Error: program token address derivation mismatch");
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    let program_token_signer_seeds: &[&[_]] = &[br"program-token", &[program_token_bump_seed]];
+
+    match instruction_data.get(0) {
+        Some(0) => {
```

Comment:
Note that the create instruction is currently unprivileged, meaning that *anybody* can grief you by creating an arbitrary token at the fixed program address.   In the real world, some authority should be checked before creating the token account